salt/utils/gitfs.py

```diff
@@ -2890,9 +2918,7 @@ def find_file(self, path, tgt_env="base", **kwargs):  # pylint: disable=W0613
         and send the path to the newly cached file
         """
         fnd = {"path": "", "rel": ""}
-        if os.path.isabs(path) or (
-            not salt.utils.stringutils.is_hex(tgt_env) and tgt_env not in self.envs()
-        ):
+        if os.path.isabs(path):
```

Comment:
@dwoz Can you explain this change? This would appear to break existing functionality which allows people to use a SHA as a saltenv in SLS files (or when using `cp.cache_file`).